build_tools/github_actions/post_build_upload.py

```diff
@@ -155,6 +155,54 @@ def upload_artifacts(args: argparse.Namespace, bucket_uri: str):
     build_dir = args.build_dir
     amdgpu_family = args.amdgpu_family
 
+    # AWS upload signatures depend on timestamps and will fail if the time differs by 5mins
+    # This will make sure the Windows machine time is synced
+    if is_windows():
+        log(f"Current time before time sync {str(datetime.now())}")
```

Comment:
This code should be placed in a helper function instead of being inlined into `upload_artifacts()`, since the implementation details are irrelevant to artifact uploading. All that `upload_artifacts()` should see is something like `sync_windows_clock()`.